src/core_plugins/region_map/public/region_map_controller.js

```diff
@@ -90,13 +90,34 @@ module.controller('KbnRegionMapController', function ($scope, $element, Private,
   async function makeKibanaMap() {
     const tmsSettings = await serviceSettings.getTMSService();
     const minMaxZoom = tmsSettings.getMinMaxZoom(false);
-    kibanaMap = new KibanaMap($element[0], minMaxZoom);
-    const url = tmsSettings.getUrl();
-    const options = tmsSettings.getTMSOptions();
-    kibanaMap.setBaseLayer({ baseLayerType: 'tms', options: { url, ...options } });
+
+    const options = { ...minMaxZoom };
+    const uiState = $scope.vis.getUiState();
+    const zoomFromUiState = parseInt(uiState.get('mapZoom'));
+    const centerFromUIState = uiState.get('mapCenter');
+    options.zoom = !isNaN(zoomFromUiState) ? zoomFromUiState : $scope.vis.type.visConfig.defaults.mapZoom;
```

Comment:
if its not in uiState you should just fallback to a fixed setting, no need to define it on visConfig